cache/md5sum.go

```diff
@@ -0,0 +1,34 @@
+package cache
+
+import (
+	"crypto/md5"
+	"fmt"
+	"io"
+	"os"
+	"path/filepath"
+)
+
+// hashSum takes in a file or directory and hashes based on filename + modified time.
```

Comment:
This hashes every directory on the path as well right?

Reply:
Sorta - it does a hashsum of all the files within a directory OR a hash on a single filename